plugins/module_utils/redfish_utils.py

```diff
@@ -145,13 +145,14 @@ def get_request(self, uri, override_headers=None):
                             force_basic_auth=basic_auth, validate_certs=False,
                             follow_redirects='all',
                             use_proxy=True, timeout=self.timeout)
-            if override_headers:
-                resp = gzip.open(BytesIO(resp.read()), 'rt', encoding='utf-8')
```

Comment:
As a note about this change, when I was refactoring this to minimize the amount of copy/paste logic for the exception handling, I determined that anything returning a gzip compressed response is automatically decompressed by `open_url`. When I was testing gzip usage on a live service, it would always fail an additional gzip decompression here due to the fact it's already decompressed.

Reply:
This seems to be due to https://github.com/ansible/ansible/commit/d58e69c82d7edd0583dd8e78d76b075c33c3151e, which was added in ansible-core 2.14. That change is kind of annoying since it's enabled by default, and disabling only works works with 2.14+.

Since this collection still supports ansible-core 2.13, I guess the best way forward is to make this code depending on the ansible-core version:

```
from ansible.module_utils.ansible_release import __version__ as ansible_version
from ansible_collections.community.general.plugins.module_utils.version import LooseVersion

if LooseVersion(ansible_version) < LooseVersion("2.14"):
    ...
```

Reply:
Ick... Okay, let me look at this more and see how to make it work for both versions...

Reply:
@felixfontein okay, I made updates here to cover older version of ansible-core.

Reply:
Thanks, looks good to me! Tell me whether you want to run some more tests, or whether I should merge the PR.

Reply:
I'm good with the changes; I did enough jumping between ansible-core versions to exercise the different paths.